src/main/java/org/mskcc/cbio/portal/scripts/ImportStructuralVariantData.java

```diff
@@ -175,27 +178,34 @@ public void importData() throws IOException, DaoException {
                         // Add structural variant
                         DaoStructuralVariant.addStructuralVariantToBulkLoader(structuralVariant);
 
-                        // Add sample to sample profile list, which is important for gene panels
-                        if (!DaoSampleProfile.sampleExistsInGeneticProfile(sample.getInternalId(), geneticProfileId) && !sampleSet.contains(sample.getStableId())) {
-                            if (genePanel != null) {
-                                DaoSampleProfile.addSampleProfile(sample.getInternalId(), geneticProfileId, GeneticProfileUtil.getGenePanelId(genePanel));
-                            } else {
-                                DaoSampleProfile.addSampleProfile(sample.getInternalId(), geneticProfileId, null);
-                            }
-                        }
-                        sampleSet.add(sample.getStableId());
-                        orderedSampleList.add(sample.getInternalId());
+                        sampleIds.add(sample.getInternalId());
                     }
                 }
             }
         }
-        DaoGeneticProfileSamples.addGeneticProfileSamples(geneticProfileId, orderedSampleList);
+        // TODO the dao methods could receive a set of sample ids (like the deletion does) instead of looping
+        if (updateMode) {
+            for (Integer sampleId : sampleIds) {
+                DaoSampleProfile.updateSampleProfile(sampleId, geneticProfileId, genePanelId);
+            }
+            DaoStructuralVariant.deleteStructuralVariants(geneticProfileId, sampleIds);
```

Comment:
not sure why the variants are deleted here, and/or why this would be after the loop. Seems wrong at first sight 🤔

Reply:
ah...I see. The bulkloader insert only happens at .flushAll()